### PR TITLE
test: add compatibility test

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 

add compatibility test

### What is changed and how it works?
1. for local test
need prepare two version's binary:
bin/dm-master.test.current
bin/dm-worker.test.current
bin/dm-master.test.provious
bin/dm-worker.test.provious
you can get them by `cp bin/dm-master.test  bin/dm-master.test.current`...
and then run `make compatibility_test` to run test case `compatibility`.
If want run other case, run `make compatibility_test CASE={case_name}`, case_name can be `all`.

2. for ci
comment `/run-compatibility-test` to trigger test
the args include `case` and `pre_commit`, `case` means test case, can be file name under `tests`, or empty(will only run case `compatibility`), or `all` to run all the case. `pre_commit` means the previous version's commit, values can be a commit hash, or `HEAD`, `HEAD~5`, `HEAD` is the head of branch master. `pre_commit`'s default value is `HEAD`.

SRE pr: https://github.com/pingcap/SRE/pull/271

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
/run-compatibility-test case=http_apis pre_commit=f1f0f6b40b6e49b834b1c975879087a0f2f21d91

**Comment by WangXiangUSTC** [comment]
/run-compatibility-test case=all pre_commit=f1f0f6b40b6e49b834b1c975879087a0f2f21d91

**Comment by WangXiangUSTC** [comment]
/run-compatibility-test case=http_apis pre_commit=f1f0f6b40b6e49b834b1c975879087a0f2f21d91

**Comment by csuzhangxc** [comment]
@amyangfei @csuzhangxc PTAL

File: tests/_utils/run_dm_master [hunk 1]
```diff
@@ -2,15 +2,21 @@
 # parameter 1: work directory
 # parameter 2: master-addr port
 # parameter 3: config file for DM-master
+# parameter 4: binary version, previous or current, can be null
```
Comment (csuzhangxc) [inline tests/_utils/run_dm_master]:
```suggestion
# parameter 4: binary version, previous or current, can be omitted
```

File: tests/_utils/run_dm_worker [hunk 2]
```diff
@@ -2,17 +2,24 @@
 # parameter 1: work directory
 # parameter 2: worker-addr port
 # parameter 3: config file for DM-worker
+# parameter 4: binary version, previous or current, can be null
```
Comment (csuzhangxc) [inline tests/_utils/run_dm_worker]: ditto

File: tests/compatibility/start.sh [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    echo "use sync_diff_inspector to check increment data first time "
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+    
+    echo "use current dm-master"
+    pkill -hup dm-master.test.previous 2>/dev/null || true
+    wait_process_exit dm-master.test.previous
+    run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml current
+    check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT
+    sleep 5
```
Comment (csuzhangxc) [inline tests/compatibility/start.sh]: Is this necessary or a little value is fine?

File: tests/README.md [hunk 4]
```diff
@@ -43,6 +47,16 @@
...
 
+### Compatibility Test
+
+1. Run Compatibility test
+   
+    Run `make compatibility_test` will run compatibility test.
+
+2. Run other test with different version's dm-master and dm-worker
+   
+   Run `make compatibility_test CASE={test_case}`, for example, `make compatibility_test CASE=http_apis` will run test `http_apis` with previous dm-master and current dm-worker, or current dm-master and previous dm-worker. The `test_case` can be `all`, and will run all the test cases.
```
Comment (csuzhangxc) [inline tests/README.md]: > **or** current dm-master and previous dm-worker

seems both of them will be run?

Comment (WangXiangUSTC) [reply tests/README.md]: yes, I update this describe

File: tests/compatibility/start.sh (hunk 3 above)
Comment (WangXiangUSTC) [reply tests/compatibility/start.sh]: decrease to 3

File: tests/_utils/run_dm_worker (hunk 2 above)
Comment (WangXiangUSTC) [reply tests/_utils/run_dm_worker]: done

**Comment by WangXiangUSTC** [comment]
@csuzhangxc @amyangfei PTAL again

**Comment by GregoryIan** [comment]
@WangXiangUSTC Cool
